apps/web/src/views/Swap/V3Swap/hooks/swapRoutingAPIArguments.ts

```diff
@@ -0,0 +1,59 @@
+import { ChainId, Currency, CurrencyAmount, TradeType } from '@pancakeswap/sdk'
+import { useMemo } from 'react'
+// import { GetQuoteArgs, INTERNAL_ROUTER_PREFERENCE_PRICE, RouterPreference } from 'state/routing/types'
+
+export interface GetQuoteArgs {
+  tokenInAddress: string
+  tokenInChainId: ChainId
+  tokenInDecimals: number
+  tokenInSymbol?: string
+  tokenOutAddress: string
+  tokenOutChainId: ChainId
+  tokenOutDecimals: number
+  tokenOutSymbol?: string
+  amount: string
+  account?: string
+  //   routerPreference: RouterPreference | typeof INTERNAL_ROUTER_PREFERENCE_PRICE
+  tradeType: TradeType
+}
+/**
+ * Returns query arguments for the Routing API query or undefined if the
+ * query should be skipped. Input arguments do not need to be memoized, as they will
+ * be destructured.
+ */
+export function useRoutingAPIArguments({
```

Comment:
I think there's no need for this hook actually? We have all the information we need within the `useBestAMMTrade` hook?